### ISSUE TITLE
Release 1.0.1 does not build

### ISSUE THREAD

**Issue by rutenkolk** [fredrikwidlund/libreactor#12]
Hello, I wanted to try out version 1.0.1 of libreactor but I can't seem to build it.

The `make hello` example does not build. The failure seems similar to #11 

```
config.status: error: in `/libdynamic-1.3.0':
config.status: error: Something went wrong bootstrapping makefile fragments
    for automatic dependency tracking.  Try re-running configure with the
    '--disable-dependency-tracking' option to at least be able to build
    the package (albeit without support for automatic dependency tracking).
See `config.log' for more details
The command '/bin/sh -c wget https://github.com/fredrikwidlund/libdynamic/releases/download/v1.3.0/libdynamic-1.3.0.tar.gz &&     tar fvxz libdynamic-1.3.0.tar.gz &&     (cd libdynamic-1.3.0/ && ./configure --prefix=/usr AR=gcc-ar NM=gcc-nm RANLIB=gcc-ranlib && make install)' returned a non-zero code: 1
make: *** [Makefile:2465: hello] Error 1
```

Since this is running in docker, I don't know what I could do differently.

Trying to install libreactor via `make install` also does not work properly on my machine. 

```
$ make install
depbase=`echo src/reactor/reactor_core.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ./libtool  --tag=CC   --mode=compile gcc -DPACKAGE_NAME=\"libreactor\" -DPACKAGE_TARNAME=\"libreactor\" -DPACKAGE_VERSION=\"1.0.1\" -DPACKAGE_STRING=\"libreactor\ 1.0.1\" -DPACKAGE_BUGREPORT=\"fredrik.widlund@gmail.com\" -DPACKAGE_URL=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -I.    -std=gnu11 -flto -fuse-linker-plugin -I./src -Wall -Wextra -Wpedantic -MT src/reactor/reactor_core.lo -MD -MP -MF $depbase.Tpo -c -o src/reactor/reactor_core.lo src/reactor/reactor_core.c &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  gcc -DPACKAGE_NAME=\"libreactor\" -DPACKAGE_TARNAME=\"libreactor\" -DPACKAGE_VERSION=\"1.0.1\" "-DPACKAGE_STRING=\"libreactor 1.0.1\"" -DPACKAGE_BUGREPORT=\"fredrik.widlund@gmail.com\" -DPACKAGE_URL=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -I. -std=gnu11 -flto -fuse-linker-plugin -I./src -Wall -Wextra -Wpedantic -MT src/reactor/reactor_core.lo -MD -MP -MF src/reactor/.deps/reactor_core.Tpo -c src/reactor/reactor_core.c  -fPIC -DPIC -o src/reactor/.libs/reactor_core.o
src/reactor/reactor_core.c:25:30: error: ‘core’ redeclared as different kind of symbol
   25 | static __thread reactor_core core = {0};
      |                              ^~~~
In file included from /usr/include/dynamic.h:30,
                 from src/reactor/reactor_core.c:8:
/usr/include/dynamic/core.h:18:30: note: previous declaration of ‘core’ was here
   18 | typedef struct core          core;
      |                              ^~~~
make: *** [Makefile:1301: src/reactor/reactor_core.lo] Error 1
```

I would really like to try out this library.

My system is Kubuntu 20.04 (Ubuntu 20.04.1 LTS) on x86/64.

**Comment by RutenkolkC** [comment]
The problem here is almost the same as #11 

using `alpine:latest` instead of edge fixes the first problem.

however there was another small thing.
On line 16 of the Dockerfile, you try to wget 
`https://github.com/fredrikwidlund/libreactor/releases/download/v1.0.0/libreactor-1.0.1.tar.gz`
which does not exist and I think should be
`https://github.com/fredrikwidlund/libreactor/releases/download/v1.0.1/libreactor-1.0.1.tar.gz` 

But still, then I get

```
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: reactor.o (symbol from plugin): in function `reactor_construct':
(.text+0x0): multiple definition of `reactor_user_default'; example/example.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: reactor_core.o (symbol from plugin): in function `reactor_core_construct':
(.text+0x0): multiple definition of `reactor_user_default'; example/example.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: reactor_fd.o (symbol from plugin): in function `reactor_fd_construct':
(.text+0x0): multiple definition of `reactor_user_default'; example/example.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: reactor_timer.o (symbol from plugin): in function `reactor_timer_construct':
(.text+0x0): multiple definition of `reactor_user_default'; example/example.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: reactor_pool.o (symbol from plugin): in function `reactor_pool_construct':
(.text+0x0): multiple definition of `reactor_user_default'; example/example.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: reactor_resolver.o (symbol from plugin): in function `reactor_resolver_construct':
(.text+0x0): multiple definition of `reactor_user_default'; example/example.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: reactor_assert.o (symbol from plugin): in function `reactor_assert_int_equal':
(.text+0x0): multiple definition of `reactor_user_default'; example/example.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: reactor_user.o (symbol from plugin): in function `reactor_user_construct':
(.text+0x0): multiple definition of `reactor_user_default'; example/example.o (symbol from plugin):(.text+0x0): first defined here
collect2: error: ld returned 1 exit status
make: *** [Makefile:1115: example/example] Error 1
The command '/bin/sh -c wget https://github.com/fredrikwidlund/libreactor/releases/download/v1.0.1/libreactor-1.0.1.tar.gz &&         tar fvxz libreactor-1.0.1.tar.gz &&         (cd libreactor-1.0.1; ./configure --prefix=/usr AR=gcc-ar NM=gcc-nm RANLIB=gcc-ranlib; make install)' returned a non-zero code: 2
```

I only tried using this version because I could not get version 2.0.0 up and running so I can undestand if this old version is no longer supported. I just wanted to update what I found out.

**Comment by fredrikwidlund** [comment]
Thanks! Yes, that is a typo, also libreactor-1.0.1 does not work with the new version of libdynamic. The readme should be updated accordingly!

**Comment by fredrikwidlund** [comment]
Btw, the error regarding "bootstrapping makefile" is likely due to spaces in the path.

**Comment by sumeetchhetri** [comment]
Please find fix for v1.0.1 build failures -- https://github.com/sumeetchhetri/libreactor/commits
